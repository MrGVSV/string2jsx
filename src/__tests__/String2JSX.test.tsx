import React from "react";
import renderer from "react-test-renderer";
import String2JSX from "../index";

describe("String2JSX", () => {
	it("renders correctly", () => {
		let conversion = [
			{
				from: /Hello/,
				to: "Goodbye",
			},
			{
				from: /, world/,
				to: <span />,
				isChild: true,
			},
			{
				from: /!/,
				to: <span>...</span>,
			},
		];
		const tree = renderer
			.create(<String2JSX map={conversion}>Hello, world!</String2JSX>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('returns default "Hello, world!"', () => {
		const tree = renderer
			.create(<String2JSX map={[]}>Hello, world!</String2JSX>)
			.toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toBe("Hello, world!");
	});

	it('maintains props', () => {
		let conversion = [
			{
				from: /Hello/,
				to: "Goodbye",
			},
			{
				from: /, world/,
				to: <span style={{ color: "red" }} />,
				isChild: true,
			},
			{
				from: /!/,
				to: <span>...</span>,
				props: {
					className: "kinda-dark-tbh",
				},
			},
		];
		const tree = renderer
			.create(<String2JSX map={conversion}>Hello, world!</String2JSX>)
			.toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('uses the given parent', () => {
        let conversion = [
            {
                from: /world/,
                to: "daddy"
            }
        ]
        const tree = renderer
			.create(<String2JSX parent={<div className="parent" />} map={conversion}>Hello, world!</String2JSX>)
			.toJSON();
        expect(tree).toMatchSnapshot();
    })
});
