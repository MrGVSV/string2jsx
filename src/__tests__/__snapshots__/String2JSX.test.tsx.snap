// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String2JSX maintains props 1`] = `
Array [
  <span>
    Goodbye
  </span>,
  <span
    style={
      Object {
        "color": "red",
      }
    }
  >
    , world
  </span>,
  <span
    className="kinda-dark-tbh"
  >
    ...
  </span>,
]
`;

exports[`String2JSX renders correctly 1`] = `
Array [
  <span>
    Goodbye
  </span>,
  <span>
    , world
  </span>,
  <span>
    ...
  </span>,
]
`;

exports[`String2JSX returns default "Hello, world!" 1`] = `"Hello, world!"`;
